### PR TITLE
Add support for Python3.10

### PR DESCRIPTION
**Pull request by skshetry**
- [ ] Waiting for appveyor/ci#3741.

Need to also update trove classifiers, tox's env, etc.

### CODE REVIEW

**Comment by skshetry** [comment]
Also so no support for 3.10 in [AppVeyor](https://www.appveyor.com/docs/windows-images-software/#python) yet.

**Comment by skshetry** [comment]
hmm, this is quiet weird:

```
/home/runner/work/pygit2/pygit2/ci/cp310/bin/python setup.py egg_info
WARNING: The wheel package is not available.
```

**Comment by dmerejkowsky** [comment]
> WARNING: The wheel package is not available.

Yeah, that can happen depending on how the virtualenv is created. A simple workaroud is to install it with pip:

```
pip install wheel
```

I also like to upgrade pip as soon as possible in my own CI scripts

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -14,6 +14,8 @@ jobs:
           python-version: 3.6
         - os: ubuntu-20.04
           python-version: 3.9
+        - os: ubuntu-20.04
+          python-version: 3.10
```
Comment (skshetry) [inline .github/workflows/tests.yml:18]:
```suggestion
          python-version: '3.10'
```

**Comment by skshetry** [comment]
Ahh, python offers 3.10 as `python-3.10.0-macos11.pkg` even though the support page says "for macOS 10.9 and later".
I'll create a separate patch for this.

**Comment by hugovk** [comment]
> * [ ]  Waiting for [Python 3.10 support appveyor/ci#3741](https://github.com/appveyor/ci/issues/3741).

Now available: https://github.com/appveyor/ci/issues/3741#issuecomment-958837889

**Pull request merged**